datawork.py

```diff
@@ -100,24 +94,22 @@ def getAllData():
         ret[device].append(data)
     return ret
 
-def getCountryVaccinePercentages():
-    vaccine_dose_counts = getCountryVaccineCounts()
-    for country in vaccine_dose_counts.keys():
-        total_doses = vaccine_dose_counts[country]["total_doses"]
-        total_percentage = 0
-        for vaccine_type in vaccine_dose_counts[country].keys():
-            if vaccine_type == "total_doses" or total_doses == 0:
-                pass
-            else:
-                percentage = vaccine_dose_counts[country][vaccine_type] * 100/(1.0 * total_doses)
-                percentage = round(percentage,2)
-                vaccine_dose_counts[country][vaccine_type] = percentage
-                total_percentage += percentage
-        del vaccine_dose_counts[country]["total_doses"]
+
+def getCountryVaccinePercentages() -> CountryVaccineSummary:
```

Comment:
this uses a pydantic model instead of a dictionary, getting rid of the "total_doses" special key. This makes the code a bit simpler